src/core/toga/widgets/canvas.py

```diff
@@ -2,334 +2,1293 @@
+    drawing with visual context.
+
+    The top left corner of the canvas must be painted at the origin of the
+    context and is sized using the rehint() method.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._children_contexts = []  # Context can have children contexts
+
+    def __repr__(self):
+        return "{}()".format(self.__class__.__name__)
+
+
+class Fill(CanvasContextMixin):
+    """A user-created :class:`Fill <Fill>` drawing object for a fill context.
+
+    A drawing object that fills the current path according to the current
+    fill rule, (each sub-path is implicitly closed before being filled).
+
+    Args:
+        color (str, optional): Color value in any valid color format,
+            default to black.
+        fill_rule (str, optional): 'nonzero' if the non-zero winding rule and
+                                   'evenodd' if the even-odd winding rule.
+        preserve (bool, optional): Preserves the path within the Context.
+
+    """
+
+    def __init__(self, color=BLACK, fill_rule="nonzero", preserve=False):
+        super().__init__()
+        self.color = parse_color(color)
+        self.fill_rule = fill_rule
+        self.preserve = preserve
+        self._children_contexts = []  # Fill context can have children contexts
+
+    def __repr__(self):
+        return "{}(color={}, fill_rule={}, preserve={})".format(
+            self.__class__.__name__, self.color, self.fill_rule, self.preserve
+        )
+
+    def __call__(self, impl, *args, **kwargs):
+        """Allow the implementation to callback the Class instance.
 
         """
-        self._impl.set_context(context)
+        impl.fill(self.color, self.fill_rule, self.preserve, *args, **kwargs)
 
-    @property
-    def on_draw(self):
-        """The callable function to draw on a Canvas
+    def modify(self, color=None, fill_rule=None, preserve=None):
+        """Modify the fill properties after it has been drawn with.
 
-        Creates a context or saves the current context, draws the operations
-        using the passed function, then renders the drawing, and finally
-        restores the saved context. The top left corner of the canvas must be
-        painted at the origin of the context and is sized using the rehint()
-        method.
+        All arguments default to the current value.
+
+        Args:
+            color (str, optional): Color value in any valid color format.
+            fill_rule (str, optional): 'nonzero' is the non-zero winding rule and
+                                       'evenodd' is the even-odd winding rule.
+            preserve (bool, optional): Preserves the path within the Context.
 
         """
-        return self._on_draw
+        if color is not None:
+            self.color = parse_color(color)
+        if fill_rule is not None:
+            self.fill_rule = fill_rule
+        if preserve is not None:
+            self.preserve = preserve
```

Comment:
The __call__ to each object requires the impl so that it can be used to call from the implementation. Did you have an idea how I might use the call for two different purposes (one for the implementation and one to modify the object)?

Reply:
I must be missing something, because I don't see how these two use cases are related. 

As I see it:
* The interface level drawing object is a container for drawing parameters (position, size, etc)
* The interface level drawing object has a `__call__` method that applies those drawing parameters to the impl backend
* The user can change the value of drawing parameters
* Those drawing parameters are reflected back on the impl backend on the next available redraw

Specifically - I can't see anywhere that `__call__` is referencing `modify`, or anywhere that `modify` is invoking `__call__` - so what am I missing?

Reply:
How would the user change all the drawing paramters at once? Would the user need to do a 
    
    rect = Rect(2, 2, 5, 5)
    rect.x = 10
    rect.y = 10
    ...
The alternative would be modifying it via rect(10, 10, 5, 5), but that would invoke the `__call__` on the object from the interface.

Reply:
Sure - but what's the use case for modifying them all at once, given that modifications are literally just a wrapper around setting individual attributes?